python/src/tests/dl/test_deep_vision_classifier.py

```diff
@@ -0,0 +1,143 @@
+from pyspark.ml.evaluation import MulticlassClassificationEvaluator
+from pyspark.sql import SparkSession
+from pyspark.sql.functions import col, udf
+from pyspark.sql.types import DoubleType
+from pytorch_lightning.callbacks import ModelCheckpoint
+
+from .test_deep_vision_model import MyDummyCallback
+
+from synapse.ml.dl import *
+
+
+@contextlib.contextmanager
+def tempdir():
+    dirpath = tempfile.mkdtemp()
+    try:
+        yield dirpath
+    finally:
+        shutil.rmtree(dirpath)
+
+
+@contextlib.contextmanager
+def local_store():
+    with tempdir() as tmp:
+        store = LocalStore(tmp)
+        yield store
+
+
+def generate_data(spark, train_folder, test_folder):
+    train_files = [
+        os.path.join(dp, f)
+        for dp, dn, filenames in os.walk(train_folder)
+        for f in filenames
+        if os.path.splitext(f)[1] == ".jpg"
+    ]
+    test_files = [
+        os.path.join(dp, f)
+        for dp, dn, filenames in os.walk(test_folder)
+        for f in filenames
+        if os.path.splitext(f)[1] == ".jpg"
+    ]
+
+    def extract_path_and_label(path):
+        num = int(path.split("/")[-1].split(".")[0].split("_")[1])
+        label = num // 81  # Assign the label
+        return (path, label)
+
+    train_df = spark.createDataFrame(
+        map(extract_path_and_label, train_files), ["image", "label"]
+    ).withColumn("label", col("label").cast(DoubleType()))
+
+    test_df = spark.createDataFrame(
+        map(extract_path_and_label, test_files), ["image", "label"]
+    ).withColumn("label", col("label").cast(DoubleType()))
+
+    return train_df, test_df
+
+
+class CallbackBackend(object):
+    def run(self, fn, args=(), kwargs={}, env={}):
+        return [fn(*args, **kwargs)] * self.num_processes()
+
+    def num_processes(self):
+        return 1
+
+
+def test_mobilenet_v2(
+    transform_row_func,
+    get_data_path,
+    read_image_and_transform_udf,
+):
+    spark = SparkSession.builder.master("local[*]").getOrCreate()
+
+    # backend = SparkBackend(
+    #     stdout=sys.stdout,
+    #     stderr=sys.stderr,
+    #     prefix_output_with_timestamp=True,
+    # )
+    ctx = CallbackBackend()
+
+    epochs = 5
+    callbacks = [
+        MyDummyCallback(epochs),
+        ModelCheckpoint(dirpath="target/mobilenet_v2/"),
+    ]
+
+    train_folder, test_folder = get_data_path
+
+    with local_store() as store:
+
+        deep_vision_classifier = DeepVisionClassifier(
+            backbone="mobilenet_v2",
+            num_layers_to_fine_tune=0,
+            store=store,
+            backend=ctx,
+            callbacks=callbacks,
+            input_shapes=[[-1, 3, 224, 224]],
+            num_classes=17,
+            feature_cols=["image"],
+            label_cols=["label"],
+            optimizer_name="adam",
+            loss_name="cross_entropy",
+            batch_size=16,
+            epochs=epochs,
+            validation=0.1,
+            verbose=1,
+            profiler=None,
```

Comment:
You might want to apply the same similplifications from the notebook to this test, perhaps minus the removal of callbacks if you find those useful for debugging